tests/generation/test_utils.py

```diff
@@ -98,9 +98,16 @@ class GenerationTesterMixin:
 
     def _get_input_ids_and_config(self, batch_size=2):
         config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
-        input_ids = inputs_dict[self.input_name]
+        input_ids = inputs_dict.pop(self.input_name)[:batch_size, :]
```

Comment:
I'd day we use `model.main_input_name` here but let's leave it for the next PR. Seems it will cause many tests to fail 😓

Reply:
let's add a todo then! :)